File: Arrays/1773. Count Items Matching a Rule.js
link = 'https://leetcode.com/problems/count-items-matching-a-rule/'


// Solution
//---------------------------------
const countMatches = (items, ruleKey, ruleValue) => {
    let count = 0
    for (let i = 0; i < items.length; i++) {
        switch (ruleKey) {
            case 'type':
                if(items[i][0]===ruleValue) count++
                break;
                
            case 'color':
                if(items[i][1]===ruleValue) count++
                break;
            case 'name':
                if(items[i][2]===ruleValue) count++
                break;
        
            default:
                break;
        }
    }
    return count
}
//---------------------------------
let result = countMatches([["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]], "color", "silver")

console.log(result)

File: Arrays/1773. Count Items Matching a Rule.py
def countMatches(items, ruleKey, ruleValue):
    count = 0
    for i in range(len(items)):
        if(ruleKey=='type'):
            if(items[i][0]==ruleValue):
                count+=1
        elif(ruleKey=='color'):
            if(items[i][1]==ruleValue):
                count+=1
        elif(ruleKey=='name'):
            if(items[i][2]==ruleValue):
                count+=1
    return count

print(countMatches([["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]], "color", "silver"))